substrate/frame/executive/src/lib.rs

```diff
@@ -424,10 +424,15 @@ where
 	pub fn try_runtime_upgrade(checks: UpgradeCheckSelect) -> Result<Weight, TryRuntimeError> {
 		let before_all_weight =
 			<AllPalletsWithSystem as BeforeAllRuntimeMigrations>::before_all_runtime_migrations();
+
 		let try_on_runtime_upgrade_weight =
-			<(COnRuntimeUpgrade, AllPalletsWithSystem) as OnRuntimeUpgrade>::try_on_runtime_upgrade(
-				checks.pre_and_post(),
-			)?;
+			<(
+				COnRuntimeUpgrade,
+				<System as frame_system::Config>::SingleBlockMigrations,
```

Comment:
Please add a test :) We already have some for this.

Reply:
Addressed in https://github.com/paritytech/polkadot-sdk/pull/9451/commits/64c75daf537c67e072782e00ca66e18fff8e3e77